src/Host/Broker/Impl/Sessions/SessionsController.cs

```diff
@@ -35,16 +35,8 @@ public class SessionsController : Controller {
                 return Task.FromResult<IActionResult>(new ApiErrorResult(BrokerApiError.NoRInterpreters));
             }
 
-            SecureString securePassword = null;
-            string password = User.FindFirst(Claims.Password)?.Value;
-            if (password != null) {
-                securePassword = new SecureString();
-                foreach (var ch in password) {
-                    securePassword.AppendChar(ch);
-                }
-            }
-
             string profilePath = User.FindFirst(Claims.RUserProfileDir)?.Value;
+            var password = User.FindFirst(Claims.Password)?.Value.ToSecureString();
```

Comment:
That's an interesting idea. We could generate GUIDs for resource name, and store those GUIDs as claims. That way, any request that _doesn't_ need the password (most of them) will never even access it, and those that do, can `CredRead` directly into a `SecureString` at the point of use, and immediately discard the string after. If I understand correctly, cred manager encrypts everything it stores with a user-specific key, so dumps etc won't contain any recoverable passwords. 

However, how do we properly clean up stored credentials, especially if broker crashes? Is there any way to set a relatively short expiration date on them?